src/pools/PoolRouter.sol

```diff
@@ -573,7 +573,7 @@ contract PoolRouter is Auth, Multicall, IPoolRouter, IPoolRouterGatewayHandler {
     /// @notice Send native tokens to the gateway for transaction payment if it's not in a multicall.
     function _pay() internal {
         if (!gateway.isBatching()) {
-            gateway.topUp{value: msg.value}();
+            gateway.payTransaction{value: msg.value}();
```

Comment:
Maybe just even simpler, `Gateway.pay()`?

Reply:
Or `Gateway.prepay()`?

Reply:
Could `pay` be confused with a more general pay? If not I'm fine

Reply:
I'm fine with sticking to `payTransaction` if you feel that is clearer!